src/main/java/ru/ayaz/Chat.java

```diff
@@ -0,0 +1,84 @@
+package ru.ayaz;
+
+
+import ru.ayaz.ru.ayaz.exceptions.InvalidNicknameException;
+import ru.ayaz.ru.ayaz.exceptions.RegistrationFailException;
+
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.PrintWriter;
+import java.net.ServerSocket;
+import java.net.Socket;
+import java.util.HashMap;
+import java.util.concurrent.ArrayBlockingQueue;
+import java.util.concurrent.BlockingQueue;
+
+public class Chat {
+
+    private ServerSocket serverSocket;
+    private UserMessageDistributor messageDistributor;
+
+    Chat() {
+        try {
+            this.serverSocket = new ServerSocket(4400);
+        } catch (IOException e) {
+            e.printStackTrace();
+        }
+    }
+
+
+    void start() {
+        startUserMessageDistributor();
+        try {
+            while (true) {
+                Socket socket = serverSocket.accept();
+                UserSocketHandler userSocketHandler = new UserSocketHandler(socket, messageDistributor);
+                User user = registerUser(userSocketHandler.getWriter(), userSocketHandler.getReader());
```

Comment:
1. Yeah, that's probably true, User should be an entity from Domain model and controller should use this model, not the other way around. The code that made me think the way I described above is:
"userSocketHandler.setNickname(user.getNickname());" and the like. If you want to use User from Controller layer, do just that, get rid of "userSocketHandler.setNickname" and the like.

2. The way it is right now, you leak implementation logic of your UserSocketHandler having things like "getWriter()" as public methods. Right now, you have the logic of reading/writing to/from socket spread out in two places - Chat and UserSocketHandler.